admin/server/projects.go

```diff
@@ -1252,25 +1377,75 @@ func (s *Server) SetProjectMemberUserRole(ctx context.Context, req *adminv1.SetP
 		return nil, status.Error(codes.PermissionDenied, "not allowed to set project member roles")
 	}
 
-	role, err := s.admin.DB.FindProjectRole(ctx, req.Role)
-	if err != nil {
-		return nil, status.Error(codes.InvalidArgument, err.Error())
-	}
-	if role.Admin && !claims.ProjectPermissions(ctx, proj.OrganizationID, proj.ID).ManageProjectAdmins {
-		return nil, status.Error(codes.PermissionDenied, "as a non-admin you are not allowed to assign an admin role")
+	user, err := s.admin.DB.FindUserByEmail(ctx, req.Email)
+	if err != nil && !errors.Is(err, database.ErrNotFound) {
+		return nil, err
 	}
 
-	user, err := s.admin.DB.FindUserByEmail(ctx, req.Email)
-	if err != nil {
-		if !errors.Is(err, database.ErrNotFound) {
+	// figure out the invite or member info
+	var invite *database.ProjectInvite
+	var member *database.ProjectMemberUser
+	if user == nil {
+		// find pending invite to get the current role
+		invite, err = s.admin.DB.FindProjectInvite(ctx, proj.ID, req.Email)
+		if err != nil {
 			return nil, err
 		}
-		// Check if there is a pending invite for this user
-		invite, err := s.admin.DB.FindProjectInvite(ctx, proj.ID, req.Email)
+	} else {
```

Comment:
Subjective, but I think the old implementation where all the invite logic was handled separately from the member logic was nicer (even if it means some duplication). With this pre-loading, there's several `if member != nil` or `if invite != nil` cases that get quite hard to understand.